### PR TITLE
 Add ability to exclude project from issue syncing/indexing

### PR DESCRIPTION
**Pull request by onass1**
Issue #9

### CODE REVIEW

File: backend/apps/github/models/repository.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ class Meta:
     has_pages = models.BooleanField(verbose_name="Has pages", default=False)
     has_projects = models.BooleanField(verbose_name="Has projects", default=False)
     has_wiki = models.BooleanField(verbose_name="Has wiki", default=False)
+    sync_issues = models.BooleanField(verbose_name="Sync Issues", default=True)
```
Comment (arkid15r) [inline backend/apps/github/models/repository.py]:
```suggestion
    sync_issues = models.BooleanField(verbose_name="Sync issues", default=True)
```

File: backend/apps/owasp/models/project.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ class ProjectType(models.TextChoices):
     pushed_at = models.DateTimeField(verbose_name="Pushed at", blank=True, null=True)
     updated_at = models.DateTimeField(verbose_name="Updated at", blank=True, null=True)
 
+    sync_issues = models.BooleanField(verbose_name="Sync Issues", default=True)
```
Comment (arkid15r) [inline backend/apps/owasp/models/project.py]:
```suggestion
    sync_issues = models.BooleanField(verbose_name="Sync issues", default=True)
```

File: backend/apps/github/common.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def sync_repository(gh_repository, organization=None, user=None):
     )
 
     # GitHub repository issues.
-    if not repository.is_archived and repository.project:
+    if not repository.is_archived and repository.project and repository.sync_issues:
```
Comment (arkid15r) [inline backend/apps/github/common.py]: You need to add the project level flag check too

File: backend/apps/github/models/repository.py [hunk 4]
```diff
@@ -106,6 +107,7 @@ def is_indexable(self):
             not self.is_archived
             and not self.is_empty
             and not self.is_template
+            and self.sync_issues
```
Comment (arkid15r) [inline backend/apps/github/models/repository.py]:
```suggestion
            and self.sync_issues
```
You need to modify issues indexing condition instead.

**Review by arkid15r — CHANGES_REQUESTED** [review]
This is what we need -- it'll allow excluding on both project and repository levels.

It seems the migration files are missing. Could you add them?

File: backend/apps/github/common.py (hunk 3 above)
Comment (naseyro) [reply backend/apps/github/common.py]: making it this way?
-----------------------------------------------------
    if not repository.is_archived and repository.project and repository.sync_issues and repository.project.sync_issues:

Comment (naseyro) [reply backend/apps/github/common.py]: I made an edit instead:
    `if not repository.is_archived and repository.project and repository.sync_issues and repository.project.is_indexable:`

also in project.py I modified the is_indexable function in Project model to:


`          return self.is_active and self.has_active_repositories and self.sync_issues`

Comment (naseyro) [reply backend/apps/github/common.py]: which way should I consider?

Comment (arkid15r) [reply backend/apps/github/common.py]: At this step let's stick with the first option that involves `sync_issues` flags only (don't touch `is_indexable`)

File: backend/apps/github/models/repository.py (hunk 4 above)
Comment (naseyro) [reply backend/apps/github/models/repository.py]: How? Isn't it already modified?

Comment (arkid15r) [reply backend/apps/github/models/repository.py]: Please see the implementation in the recent commit.

**Review by arkid15r — APPROVED** [review]
Merging this in, thanks for the contribution @onass1!

**Pull request merged**